.github/workflows/build-and-test.yml

```diff
@@ -112,6 +112,22 @@ jobs:
           ~/actionutils.sh wait_for_osds 3
           sudo microceph.ceph -s
 
+      - name: Add and relate RadosGW charm
+        run: |
+          set -eux
+          juju deploy -m admin/controller ch:ceph-radosgw
+          sleep 120
+          juju integrate -m admin/controller ceph-radosgw microceph
+          sleep 420
+          cnt=$(juju ssh microceph/0 -- "sudo microceph.ceph osd pool ls | wc -l")
+          if [[ "$cnt" -ge 0 ]]; then
+            echo "radosgw unit is active"
+          else
+            echo "radosgw not active after timeout"
+            sudo microceph.ceph -s
+            exit 1
+          fi
```

Comment:
I believe the test for having more than 0 pools doesn't work because we'll have the `.mgr` pool from the start. Also it would be great if we don't sleep a fixed amount of time but check periodically. 

May I suggest running something like the below loop to periodically check for the number of pools. I *think* expecting at least 4 pools would be an indicator for radosgw.

```
    expect 4
    res=0
    for i in $(seq 1 20); do
        res=$( ( juju ssh microceph/0 -- "sudo microceph.ceph osd pool ls | wc -l" ) )
        if [[ $res -ge $expect ]] ; then
            echo "Found >= ${expect} pools"
            break
        else
            echo -n '.'
            sleep 30
        fi
    done
    if [[ $res -lt $expect ]] ; then
        echo "Never reached ${expect} pools"
        juju ssh microceph/0 -- "sudo microceph.ceph osd pool ls"
        juju ssh microceph/0 -- "sudo microceph.ceph -s"
        exit 1
    fi
```

Reply:
Done.